WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostPagerActivity.kt

```diff
@@ -0,0 +1,1211 @@
+        // We interrupted the normal flow to show the overly, we now need to rerun these methods on a dismiss action
+        loadPosts(blogId, postId)
+        backFromLogin = false
+    }
+
+    @Suppress("NestedBlockDepth", "MagicNumber")
+    private fun finishDeepLinkRequest(action: String, uri: Uri) {
+        var interceptType = InterceptType.READER_BLOG
+        var blogIdentifier: String? = null // can be an id or a slug
+        var postIdentifier: String? = null // can be an id or a slug
+
+        interceptedUri = uri.toString()
+
+        val segments = uri.pathSegments
+
+        // Handled URLs look like this: http[s]://wordpress.com/read/feeds/{feedId}/posts/{feedItemId}
+        // with the first segment being 'read'.
+        if (segments != null) {
+            // Builds stripped URI for tracking purposes
+            val wrappedUri = UriWrapper(uri)
+            if (segments[0] == "read") {
+                if (segments.size > 2) {
+                    blogIdentifier = segments[2]
+
+                    if (segments[1] == "blogs") {
+                        interceptType = InterceptType.READER_BLOG
+                    } else if (segments[1] == "feeds") {
+                        interceptType = InterceptType.READER_FEED
+                        isFeed = true
+                    }
+                }
+
+                if (segments.size > 4 && segments[3] == "posts") {
+                    postIdentifier = segments[4]
+                }
+
+                parseFragment(uri)
+                deepLinkTrackingUtils.track(action, OpenInReader(wrappedUri), wrappedUri)
+                showPost(interceptType, blogIdentifier, postIdentifier)
+                return
+            } else if (segments.size >= 4) {
+                blogIdentifier = uri.host
+                try {
+                    postIdentifier = URLEncoder.encode(segments[3], "UTF-8")
+                } catch (e: UnsupportedEncodingException) {
+                    AppLog.e(AppLog.T.READER, e)
+                    ToastUtils.showToast(this, R.string.error_generic)
+                }
+
+                parseFragment(uri)
+                detectLike(uri)
+
+                interceptType = InterceptType.WPCOM_POST_SLUG
+                deepLinkTrackingUtils.track(action, OpenInReader(wrappedUri), wrappedUri)
+                showPost(interceptType, blogIdentifier, postIdentifier)
+                return
+            }
+        }
+
+        // at this point, just show the entry screen
+        val intent = Intent(this, WPLaunchActivity::class.java)
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK)
+        startActivity(intent)
+    }
+
+    @Suppress("MagicNumber")
+    private fun showPost(
+        interceptType: InterceptType,
+        blogIdentifier: String?,
+        postIdentifier: String?
+    ) {
+        if (!TextUtils.isEmpty(blogIdentifier) && !TextUtils.isEmpty(postIdentifier)) {
```

Comment:
```suggestion
        if (!blogIdentifier.isNullOrEmpty() && !postIdentifier.isNullOrEmpty()) {
```
Just a nitpick to avoid using Java methods

Reply:
Good call! I made that change in [1f0f899](https://github.com/wordpress-mobile/WordPress-Android/pull/21822/commits/1f0f8994e844497ab0da17a5206e957b7a522ea8)